src/org/thoughtcrime/securesms/permissions/Permissions.java

```diff
@@ -100,6 +102,20 @@ public PermissionsBuilder ifNecessary(boolean condition) {
       return this;
     }
 
+    public PermissionsBuilder alwaysGrantOnSdk30() {
```

Comment:
I would just have a generic function `alwaysGrantOnSdk(int version)` and then call `Permissions.alwaysGrantOnSdk(30)` etc.

Reply:
yeah, let's leave this as is for now. it looks more explicit that 33/30 is no typo in a parameter :)